File: Java/Data Structures & Algorithms Module - Java/Week8/Revision/Week7/RemoveX.java
/*
RemoveX.java
Billy Cussen
19/11/2020
*/
public class RemoveX{

    public static void main(String[] args){

        System.out.println("Remove X: "+removeX("BxixlxlxyX CxuXsXsXeXnx"));
        System.out.println("Remove X Recursion: "+removeXRecursion("BxixlxlxyX CxuXsXsXeXnx"));
    }

    public static String removeX(String word){
        StringBuffer sb = new StringBuffer();
        for(int i = 0; i < word.length();i++){
            if(word.charAt(i)!='x' && word.charAt(i)!='X'){
                sb.append(word.charAt(i));
            }
        }
        return sb.toString();
    }

    public static String removeXRecursion(String word){
        if(word.length()<=0){
            return word;
        }

        if(word.charAt(0)!='x'&&word.charAt(0)!='X'){
            return word.charAt(0) + removeXRecursion(word.substring(1,word.length()));
        } else{
            return removeXRecursion(word.substring(1,word.length()));
        }
    }

}

File: Python/Data Structures & Algorithms Module - Python/Revision/Recursion/RemoveX.py
"""
RemoveX.py
Billy Cussen
13/01/2021
"""

def removeX(word):
    if len(word) <= 0:
        return word
    if word[0] == 'x' or word[0] == 'X':
        return removeX(word[1:len(word)])
    else:
        return word[0] + removeX(word[1:len(word)])

print(removeX("BxiXlxlXyx"))